development/webpack/utils/plugins/ManifestPlugin/helpers.ts

```diff
@@ -29,28 +38,72 @@ export function transformManifest(args: { lockdown: boolean; test: boolean }) {
     transforms.push(removeLockdown);
   }
 
-  function addTabsPermission(browserManifest: chrome.runtime.Manifest) {
-    if (browserManifest.permissions) {
-      if (browserManifest.permissions.includes('tabs')) {
-        throw new Error(
-          "manifest contains 'tabs' already; this transform should be removed.",
+  /**
+   * This function sets predefined flags in the manifest's _flags property
+   * that are stored in the .manifest-flags.json file.
+   *
+   * @param browserManifest - The Chrome extension manifest object to modify
+   */
+  function addManifestFlags(browserManifest: chrome.runtime.Manifest): void {
+    let manifestFlags;
+
+    if (manifestOverridesPath) {
+      try {
+        const fs = require('node:fs');
+        const path = require('node:path');
+        const manifestFlagsContent = fs.readFileSync(
+          path.resolve(process.cwd(), manifestOverridesPath),
+          'utf8',
         );
+        manifestFlags = JSON.parse(manifestFlagsContent);
+      } catch (error: unknown) {
+        if (
+          error instanceof Error &&
+          'code' in error &&
+          error.code === 'ENOENT'
+        ) {
+          // Only throw if ENOENT and manifestOverridesPath was provided
+          throw new Error(
+            `Manifest override file not found: ${manifestOverridesPath}`,
+          );
+        }
       }
+    }
+
+    if (manifestFlags) {
+      merge(browserManifest, manifestFlags);
+    }
+  }
+
+  if (isDevelopment) {
+    // Add manifest flags only for development builds
+    transforms.push(addManifestFlags);
+  }
+
+  function addTabsPermission(browserManifest: chrome.runtime.Manifest) {
+    if (browserManifest.permissions?.includes('tabs')) {
+      throw new Error(
+        "manifest contains 'tabs' already; this transform should be removed.",
+      );
+    }
```

Comment:
Why is this moved? The existence check for `browserManifest.permissions` is now done twice, where before it was only checked one time.

Reply:
🕵🏻‍♂️ good catch, reverted 0380cf085de9f0426178f63b5ac8f58e741ec9ce